### ISSUE TITLE
Can I pass any data with RouteAction?

### ISSUE THREAD

**Issue by nikans** [ReSwift/ReSwift-Router#63]
With ReSwift `StandardAction` I can pass some arbitrary payload. Looks like I can't do it with the help of `SetRouteAction`, because it only takes an array of `RouteElementIdentifier`s which are, unfortunately, strings.
I think there's an opportunity for major overhaul, because routes for parametrized controllers are effectually impossible. Also string route identifiers stink.

**Comment by JamesPerlman** [comment]
I tried creating an extension called RoutableEnumerable which lets you SetRouteAction([SomeEnum.someCase]).  Thanks to Swift enums we can have associated values.  The problem is with stacking enums of different types, it throws a compiler error that says the SetRouteAction cannot be initialized with an array of type [Any].

My peer pointed out that we could simply store all routable states in a single enum to avoid the typecasting issue (a common protocol for each enum didn't help either) - but our app is enormous, and having all routable states in a single enum would mean we could only have one gigantic inflated Routable.

Long story short, I believe this is an abuse of Redux.  Personally, I believe you should set your state in one ViewController (or whatever class/entity), then dispatch your SetRouteAction, and read the state from the target entity/class/viewcontroller you wish to "pass" data to.

Check out the example project.  They use this pattern when using Bookmarks.

Hope that helps.

**Comment by DivineDominion** [comment]
Parameters to support nested detail views, for example?

    /user/Foo/photos/12345/user/Bar
    Foo profile
              Her pirctures with ID 12345
                           Bar user profile (linked from picture)

That should be doable, but passing the 2 user type references and the photo item with the state, not so much. Is it the latter you want to achieve?

**Comment by AlexanderBollbach** [comment]
@JamesPerlman I find the idea of setting state (global) and then pushing a route and then reading from the state in the next 'component' (for lack of a better term on iOS) plausible in the context of the 'keep everything in the store' philosophy.  But one thing I've noticed on web is that people definitely use react 'props' to pass 'transient' data they get from a routing action.  Usually a component will be created with as little data as possible, e.g. a 'todo id' and then it will select that todo from the store.  My feeling is that this prevents the state shape from becoming too coupled on the navigation state.

To that end I've been using a stringly typed schema e.g. `["todos", "todo:{42}"]` for passing navigation data to build up a 'nav tree'.  The data is associated with the current navigation state (through the navigationState) and not muddied together with the domain state.
